### DISCUSSION TITLE
Axisymmetric coordinates postprocessing

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#24117] [Q&A General] [answered]
Hello everyone,

I had two questions regarding the axisymmetric coordinates.

1. Let's say that I solved my problem in RZ coordinates. Now I want to visualize my 2D data in 3D. What is the most convenient way to do it?
2. I have a variable which is called "c". I want to integrate c over the 3D volume, not the 2D surface. Is there any available kernel for that? This volume integration would be an important part of my work.

Thank you,
Ali

**Comment by jiangwen84** [comment, marked as answer]
1. You can use Preview, see Rotational Extrusion filter.
2. If you are using PostProcessor, it will automatically account for coordinate system. It does volume integral.

**Reply by hugary1995** [reply]
Not Preview, but Paraview :)

You may need to "Extract Surface" before applying the extrusion filter.

**Reply by jiangwen84** [reply]
And "Transform" filter :-)

**Reply by alimostafavi24** [reply]
1. I used Paraview. Firstly, I needed to use "Extract Surface" and then  "Rotational Extrusion".
2. This is part of my input file and I use RZ coordinates. So this one calculates the c volume integral on the domain. Correct?
```
[Postprocessors]
  [./c_integral]
    type = ElementIntegralVariablePostprocessor
    variable = c
  [../]
[]
```

**Reply by GiudGiud** [reply]
Correct. This is over the entire mesh, and in 2D RZ it corresponds to a 3D volume integral because the local ~~element~~ Qp volumes are multiplicated by 2 pi r

**Reply by hugary1995** [reply]
No, it doesn't work by multiplying element volumes by 2 pi r, which would yield the wrong answer. There's a coordinate transformation Jacobian which is r.

**Reply by alimostafavi24** [reply]
Ok, so because of the Jacobian coordinate transformation, the volume integral would be correctly calculated? Here is my results for c integration (order parameter in the phase field), which is supposed to be conserved. I don't know why I get the oscillation!


![untitled](https://user-images.githubusercontent.com/119982603/233220964-fb3c5c22-4149-43ae-acdf-6bac1aa724dc.jpg)

Thanks,
Ali

**Reply by GiudGiud** [reply]
what is the X axis here ?

**Reply by alimostafavi24** [reply]
Y axis is c integral, and X axis is time.

**Reply by jiangwen84** [reply]
What is your BC? Are you running a C-H equation? It only conserves when zero flux BC is applied for chemical potential variable.

**Reply by alimostafavi24** [reply]
I am running coupled cahn hilliard navier stokes equation. Chemical potential's BC is neumann BC on all 4 boundaries.

**Reply by jiangwen84** [reply]
Let us try a simple CH equation,  without using NS.

**Comment by alimostafavi24** [comment]
Thank you very much (@jiangwen84 @hugary1995 @GiudGiud )!
